KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/AuthenticationModule.java

```diff
@@ -91,7 +91,12 @@ private void authenticate(String entryId, int partnerId, String token, KalturaEn
             try {
                 logger.debug("(" + entryId + ") Starting authentication process");
                 if (Boolean.TRUE.equals(KalturaEntryDataPersistence.getPropertyByEntry(entryId, Constants.KALTURA_ENTRY_AUTHENTICATION_ERROR_FLAG))) {
-                    throw new Exception("(" + entryId + ") Authentication Error Flag is up!");
+                    //throw new Exception("(" + entryId + ") Authentication Error Flag is up!");
+                    long currentTime = System.currentTimeMillis();
+                    long errorTime = (long)KalturaEntryDataPersistence.getPropertyByEntry(entryId, Constants.KALTURA_ENTRY_AUTHENTICATION_ERROR_TIME);
+                    String errMsg = (String)KalturaEntryDataPersistence.getPropertyByEntry(entryId, Constants.KALTURA_ENTRY_AUTHENTICATION_ERROR_MSG);
+                    throw new Exception("Error occurred [" + (currentTime - errorTime)/1000 + "] seconds ago: " + errMsg +
+                            ". Connection is blocked for up to " + (Constants.KALTURA_PERSISTENCE_DATA_MIN_ENTRY_TIME / 1000) + " seconds");
```

Comment:
where do you write the url that is currently used to connect?
Maybe better to write: Connection blocked due to previous authentication failure. Try reconnect in X sec.
where X =  Constants.KALTURA_PERSISTENCE_DATA_MIN_ENTRY_TIME - (currentTime - errorTime)/1000